worldedit-core/src/main/java/com/sk89q/worldedit/world/biome/BiomeTypes.java

```diff
@@ -297,10 +300,18 @@ public static BiomeType register(final BiomeType biome) {
     }
 
     public static BiomeType getLegacy(int legacyId) {
-        for (BiomeType type : values()) {
-            if (type.getLegacyId() == legacyId) {
-                return type;
+        class BiomeTypeIdCache {
+            private static final List<BiomeType> byId = create();
+
+            private static List<BiomeType> create() {
+                BiomeType[] array = values().toArray(new BiomeType[0]);
+                Arrays.sort(array, Comparator.comparing(BiomeType::getLegacyId));
+                return List.of(array);
             }
+
+        }
+        if (legacyId >= 0 && legacyId < BiomeTypeIdCache.byId.size()) {
+            return BiomeTypeIdCache.byId.get(legacyId);
```

Comment:
Any specific reason on using a List instead of an array?

Reply:
Not really, the list is immutable and its backing array is marked stable which might help performance in some cases, but it likely doesn't matter.

Reply:
Yeah I think any performance drops from using a list should be compiled away given it's a static final list, so it should just act as an array in runtime